rfq/oracle.go

```diff
@@ -60,6 +61,31 @@ type OracleBidResponse struct {
 	Err *OracleError
 }
 
+// ParsePriceOracleAddress parses a price oracle service address string and
+// returns a URL type instance.
+func ParsePriceOracleAddress(addrStr string) (*url.URL, error) {
```

Comment:
If we're keeping the custom URI, then we can add a type alias here to distinguish things a bit more: 
```
type OracleAddr = url.URL
```